ui/app/components/clients/activity.ts

```diff
@@ -65,18 +65,18 @@ export default class ClientsActivityComponent extends Component<Args> {
       return activity.byMonth;
     }
     const namespaceData = activity.byMonth
-      .map((m) => m.namespaces_by_key[namespace as keyof typeof m.namespaces_by_key])
+      .map((m) => m.namespaces_by_key[namespace])
       .filter((d) => d !== undefined);
 
     if (!mountPath) {
-      return namespaceData.length === 0 ? undefined : namespaceData;
+      return !namespaceData ? [] : namespaceData;
```

Comment:
It seemed reasonable that this getter `filteredActivityByMonth` should consistently return an array, eh? To accommodate this I updated the only falsey check below (line 125) from `!this.byMonthActivityData)` to `this.byMonthActivityData.length === 0`

Reply:
Namespace data will always be truthy since it is an empty array if filtered to no data. So if we're confident that `activity.byMonth` will always be defined we can just return `namespaceData` here, or `return namespaceData || []` if we want to add a `?` on line 67 for extra safety: `const namespaceData = activity.byMonth?.map(...)`

Reply:
Ah yeah - I got mixed up on some of these getter returns during the refactor, thank you!

Yes, byMonth will always be some sort of array